firestore/google/cloud/firestore_v1beta1/document.py

```diff
@@ -423,9 +428,38 @@ def get(self, field_paths=None, transaction=None):
         if isinstance(field_paths, six.string_types):
             raise ValueError(
                 "'field_paths' must be a sequence of paths, not a string.")
-        snapshot_generator = self._client.get_all(
-            [self], field_paths=field_paths, transaction=transaction)
-        return _consume_single_get(snapshot_generator)
+
+        if field_paths is not None:
+            mask = common_pb2.DocumentMask(field_paths=sorted(field_paths))
+        else:
+            mask = None
+
+        firestore_api = self._client._firestore_api
+        timestamp = _make_timestamp()
+        try:
+            document_pb = firestore_api.get_document(
+                self._document_path,
+                mask=mask,
+                transaction=_helpers.get_transaction_id(transaction),
+                metadata=self._client._rpc_metadata)
+        except exceptions.NotFound:
+            data = None
+            exists = False
+            create_time = None
+            update_time = None
+        else:
+            data = _helpers.decode_dict(document_pb.fields, self._client)
+            exists = True
+            create_time = document_pb.create_time
+            update_time = document_pb.update_time
+
+        return DocumentSnapshot(
+            reference=self,
+            data=data,
+            exists=exists,
+            read_time=timestamp,  # No server read_time available
```

Comment:
The older "batch get" API call had an actual `read_time` on the response, but `GetDocument` just returns `Document`, which has no such field.  So, we compute our own timestamp and use that.

Reply:
I think that it should probably be None instead of a client-generated time.  Or just remove the field entirely from Document, with attendant bw compat issues.

Reply:
DocumentSnapshot, not Document.

Reply:
OK.  I felt a little queasy about generating the timestamp myself.

Reply:
Ah, I remember now -- there is a system test which tries to use `read_time`:  I will have to adjust it.